src/ansys/units/systems.py

```diff
@@ -49,13 +55,12 @@ def __init__(self, name: str = None, base_units: list = None, unit_sys: str = No
             if not isinstance(unit, ansunits.Unit):
                 unit = ansunits.Unit(unit)
 
-            if unit.name not in ansunits._fundamental_units:
-                raise UnitSystemError.NOT_FUNDAMENTAL(unit)
+            if unit.name not in ansunits._base_units:
+                raise UnitSystemError.NOT_BASE_UNIT(unit)
 
-            if hasattr(self, f"_{unit._type.lower()}"):
-                raise UnitSystemError.UNIT_TYPE(unit)
+            unit_type = [x.name for x in unit.dimensions.dimensions.keys()][0]
```

Comment:
OK, so dumb question but does the `_base_units` entry for the required unit not already contain the required `unit_type` string loaded from the yaml configuration?  If so, can you not just use that here?

Reply:
Playing around a little I would think this is what you want for the for loop bit:
```
        for unit in self._base_units:
            if not isinstance(unit, ansunits.Unit):
                unit = ansunits.Unit(unit)

            try:
                base_info = ansunits._base_units[unit.name]
            except KeyError:
                raise UnitSystemError.NOT_BASE_UNIT(unit)

            try:
                unit_type = base_info['type']
            except KeyError:
                raise UnitSystemError.NOT_BASE_UNIT(unit)

            setattr(self, f"_{unit_type}", unit)
```